include/gridtools/fn/unstructured.hpp

```diff
@@ -84,7 +84,8 @@ namespace gridtools::fn {
         template <class Tag, class Ptr, class Strides, class Domain, class Conn, class Offset>
         GT_FUNCTION constexpr auto horizontal_shift(iterator<Tag, Ptr, Strides, Domain> const &it, Conn, Offset) {
             auto const &table = host_device::at_key<Conn>(it.m_domain.m_tables);
-            auto new_index = it.m_index == -1 ? -1 : get<Offset::value>(neighbor_table::neighbors(table, it.m_index));
+            const auto neighbor = get<Offset::value>(neighbor_table::neighbors(table, it.m_index));
```

Comment:
This gives an out-of-bounds access if `it.m_index` is indeed `-1`, doesn’t it? (Which is a functional change to the previous code, so it makes sense that the compiler can apply different optimizations, e.g., load the neighbor table independently of the value of `it.m_index`).

Reply:
Yes that's true so it makes sense. Is there any case that the `m_index` check can be avoided? I'm wondering what is special with the nabla kernels that they were the only ones that benefited from this change. Also if the total runtime of the nabla kernels in the production case is small it's probably not worth to try and improve this part.
@havogt since this change alters the behavior and is possibly problematic should there be a test in the CI for a breaking case?

Reply:
Right, I guess we should go for the compile time `has_skip_value` check to achieve this and possibly even better performance?

Reply:
Yes, I can try it out and launch the `perftests` CI to check